config/v1/types_cluster_version.go

```diff
@@ -707,3 +726,89 @@ type ClusterVersionList struct {
+	// to apply the update. Note that in that case no bundle will be used.
+	//
+	// If explicitly set to true then the desiredUpdate.version and desiredUpdate.image fields
+	// will be ignored. Instead the cluster will loop for the update bundle and will use to
+	// determine the version, pin and pre-loaded the images and appy the update.
+	//
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// monitor indicates if the cluster should monitor device plug events to automatically
+	// detect when an update bundle is available. The default is false.
+	//
+	// +optional
+	Monitor bool `json:"monitor"`
+
+	// file indicates the location of the update bundle when the monitor fields is false. It can
+	// be an absolute file path name or an abosolute block device name.
+	//
+	// +optional
+	File string `json:"file"`
+
+	// node contains the name of the node where the update bundle is available. This is
+	// optional, and by default the cluster will look for the bundle in all the nodes.
+	//
+	// +optional
+	Node string `json:"node"`
+
+	// digest is an optional digest of the update bundle. If set then the cluster will ensure
+	// that the bundle is used only if it matches this digest. This is intended to prevent use
+	// of a the wrong bundle in the wrong cluster, or use of a perfectly fine bundle in the
+	// wrong cluster.
+	//
+	// The format is be identifier of the digest algorithm followed by a colon and the digest
+	// value.  Currently the only supported digest algorithm is sha256, so the value shold
+	// be something like sha256:1ef...03a.
+	//
+	// +optional
+	Digest string `json:"digest"`
+}
+
+// ReleaseBundle contains the details of an update bundle automatically detected by the cluster
+// or manually specified by the admnistrator.
+type ReleaseBundle struct {
+	// file indicates the location of the bundle. It will be automatically populated by the
+	// cluster when monitoring is enabled via the spec.desiredUpdate.bundle.monitor field and
+	// a bundle has been detected.
+	//
+	// If monitoring is disabled then it will be copied from the spec.desiredUpdate.bundlefile
+	// field.
+	//
+	// +optional
+	File string `json:"file"`
+
+	// nodes contains the current status of the bundle processing in each node of the cluster.
+	// Keys are the names of the nodes.
+	//
+	// +optional
+	Nodes map[string]NodeBundleStatus
+}
+
+// NodeBundleStatus contains the current status of the update bundle processing for a specific node
+// of the cluster.
+type NodeBundleStatus struct {
+	// extracted indicates if the upgrade bundle has already been extracted to a temporary
+	// directory in the node.
+	//
+	// +required
+	Extracted bool `json:"extracted"`
+
+	// loaded indicates if the images of the upgrade bundle have been copied to the containers
+	// storage directory of the node.
+	//
+	// +required
+	Loaded bool `json:"loaded"`
+
+	// metadata is the content of the metadata.json file that is part of the bundle.
```

Comment:
How big is the metadata.json file typically? What kind of information does it store?